### PR TITLE
[react-table]Add react-table

### PR DESCRIPTION
**Pull request by royxue**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

### CODE REVIEW

File: types/react-table/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,509 @@
+// Type definitions for react-table 6.5
+// Project: https://github.com/react-tools/react-table#readme
```
Comment (plantain-00) [inline types/react-table/index.d.ts]: `#readme` can be removed here

**Review by plantain-00 — APPROVED** [review]

File: types/react-table/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,509 @@
...
+         * @example 'Footer Name'
+         * @example ({data, column}) => <div>Footer Name</div>,
+         */
+        footer: TableCellRenderer;
+
+        /**
+         * Default: ''
+         * Set the classname of the `td` element of the column's footer
+         */
+        footerClassName: '';
```
Comment (RyanCavanaugh) [inline types/react-table/index.d.ts]: This implies `''` is the *only* legal string value!

File: types/react-table/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,509 @@
...
+         * @example 'Header Name'
+         * @example ({data, column}) => <div>Header Name</div>,
+         */
+        header: TableCellRenderer;
+
+        /**
+         * Set the classname of the `th` element of the column
+         * @default ''
+         */
+        headerClassName: '';
```
Comment (RyanCavanaugh) [inline types/react-table/index.d.ts]: Same here - all uses of `''` as a type need to be changed

**Review by RyanCavanaugh — CHANGES_REQUESTED** [review]

**Comment by royxue** [comment]
@RyanCavanaugh 
Thanks for your comments, resolved

**Pull request merged**